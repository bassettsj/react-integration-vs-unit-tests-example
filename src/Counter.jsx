import React, { useState } from "react"
import CounterBtn from "./CounterBtn"

class Counter extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      count: 0
    }
  }

  increment = () => {
    this.setState({
      count: this.state.count + 1
    })
  }

  decrement = () => {
    this.setState({
      count: this.state.count - 1
    })
  }
  render() {
    const {
      increment,
      decrement,
      state: { count }
    } = this
    return (
      <>
        <p>Times clicked: {count}</p>
        <CounterBtn onClick={increment}>+</CounterBtn>
        <CounterBtn onClick={decrement}>-</CounterBtn>
      </>
    )
  }
}

export default Counter
