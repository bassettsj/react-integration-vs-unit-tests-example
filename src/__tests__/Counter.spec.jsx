import { shallow } from "enzyme"
import React from "react"
import Counter from "../Counter"

describe("<Counter />", () => {
  it("should should render", () => {
    expect(shallow(<Counter />)).toMatchSnapshot()
  })

  it("should increment", () => {
    const instance = shallow(<Counter />).instance()
    instance.increment()
    expect(instance).toHaveProperty("state.count", 1)
  })

  it("should decrement", () => {
    const instance = shallow(<Counter />).instance()
    instance.decrement()
    expect(instance).toHaveProperty("state.count", -1)
  })
})
