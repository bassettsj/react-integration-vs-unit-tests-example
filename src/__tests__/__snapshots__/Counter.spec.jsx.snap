// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> should should render 1`] = `ShallowWrapper {}`;
